sdks/java/core/src/main/java/org/apache/beam/sdk/util/DoFnInfo.java

```diff
@@ -47,34 +45,30 @@
       WindowingStrategy<?, ?> windowingStrategy,
       Iterable<PCollectionView<?>> sideInputViews,
       Coder<InputT> inputCoder,
-      long mainOutput,
-      Map<Long, TupleTag<?>> outputMap) {
+      TupleTag<OutputT> mainOutput) {
     return new DoFnInfo<>(
-        doFn, windowingStrategy, sideInputViews, inputCoder, mainOutput, outputMap);
+        doFn, windowingStrategy, sideInputViews, inputCoder, mainOutput);
```

Comment:
Don't we still use this for Dataflow in the NonFnApi mode (which passes connections by index)?

Reply:
No, we actually use the names of each output instead of the index except for a few transforms which have always assumed that there is exactly one output so the name and index were irrelevant.

Reply:
I was actually thinking about the CloudMapTask items which reference connections by index. (Or has the non-FnApi code been augmented to inspect the FnApi protos?)

Reply:
The non FnApi code uses the names directly, it doesn't need to look at the protos since the names match the main output within the DoFnInfo or the serialized tuple tags within the users code contained within the DoFn.

Reply:
Ah, yes. Thanks for clarifying. LGTM pending tests all passing.